File: back/kotlin/app/core/repository/GameRepository.kt
interface GameRepository {
    fun initializeGame(): Flow<Result<GameType>>
    fun getGame(): Result<GameType>
    fun makeMove(row: Int, col: Int): Result<GameType>
    fun getNextPlayer(): Result<PlayerType>
    fun checkForWinner(): Result<PlayerType>
    fun checkForDraw(): Result<Boolean>
}

File: back/swift/app/core/repository/GameRepository.swift
import Foundation

protocol GameRepository {
    func initializeGame() -> Result<GameType, Error>
    func getGame() -> Result<GameType, Error>
    func makeMove(row: Int, col: Int) -> Result<GameType, Error>
    func getNextPlayer() -> Result<PlayerType, Error>
    func checkForWinner() -> Result<PlayerType, Error>
    func checkForDraw() -> Result<Bool, Error>
}